src/mame/video/segasaturn_vdp1.h

```diff
@@ -0,0 +1,286 @@
+// license:LGPL-2.1+
+// copyright-holders:David Haywood, Angelo Salese, Olivier Galibert, Mariusz Wojcieszek, R. Belmont
+/**************************************************************************************************
+
+Sega Saturn / ST-V - VDP1
+
+**************************************************************************************************/
+
+#ifndef MAME_VIDEO_SATURN_VDP1_H
+#define MAME_VIDEO_SATURN_VDP1_H
+
+#pragma once
+
+
+#define RGB_R(_color)   (_color & 0x1f)
+#define RGB_G(_color)   ((_color >> 5) & 0x1f)
+#define RGB_B(_color)   ((_color >> 10) & 0x1f)
+
+#define SWAP_INT32(_a,_b) \
+	{ \
+		int32_t t; \
+		t = _a; \
+		_a = _b; \
+		_b = t; \
+	}
+
+#define SWAP_INT32PTR(_p1, _p2) \
+	{ \
+		int32_t *p; \
+		p = _p1; \
+		_p1 = _p2; \
+		_p2 = p; \
+	}
```

Comment:
These shouldn’t be necessary at all – you can just use `std::swap` to exchange things.